src/FishyFlip/OpenGraphParser.cs

```
@@ -117,6 +117,12 @@ public void Dispose()
             urlEmbed = tag1;
         }
 
+        // These three fields are all required, according to the lexicon.
+        if (urlEmbed == null || title == null || description == null)
+        {
+            return null;
```

If it fails to find any one of these properties, it should throw an Exception. Otherwise you'll end up with a null object with no idea why it failed.

---

`GenerateEmbedExternal` already returns `null` to represent missing or incorrect OpenGraph tags, and `bskycli` prints an error message in such case.